src/main/java/me/zeroeightsix/kami/module/modules/client/Capes.kt

```diff
@@ -22,100 +42,211 @@ import javax.net.ssl.HttpsURLConnection
+        GlStateManager.color(color.r / 255.0f, color.g / 255.0f, color.b / 255.0f, 1.0f)
+        renderer.bindTexture(texture.location)
+        GlStateManager.pushMatrix()
+        GlStateManager.translate(0.0f, 0.0f, 0.125f)
+
+        val interpolatedPos = EntityUtils.getInterpolatedPos(player, partialTicks)
+        val relativePosX = player.prevChasingPosX + (player.chasingPosX - player.prevChasingPosX) * partialTicks - interpolatedPos.x
+        val relativePosY = player.prevChasingPosY + (player.chasingPosY - player.prevChasingPosY) * partialTicks - interpolatedPos.y
+        val relativePosZ = player.prevChasingPosZ + (player.chasingPosZ - player.prevChasingPosZ) * partialTicks - interpolatedPos.z
+
+        val yawOffset = Math.toRadians(player.prevRenderYawOffset + (player.renderYawOffset - player.prevRenderYawOffset) * partialTicks.toDouble())
+        val relativeX = sin(yawOffset)
+        val relativeZ = -cos(yawOffset)
+
+        var angle1 = MathHelper.clamp(relativePosY.toFloat() * 10.0f, -6.0f, 32.0f)
+
+        var angle2 = (relativePosX * relativeX + relativePosZ * relativeZ).toFloat() * 100.0f
+        val angle3 = (relativePosX * relativeZ - relativePosZ * relativeX).toFloat() * 100.0f
+        if (angle2 < 0.0f) {
+            angle2 = 0.0f
+        }
+
+        val cameraYaw = player.prevCameraYaw + (player.cameraYaw - player.prevCameraYaw) * partialTicks
+        val walkedDist = player.prevDistanceWalkedModified + (player.distanceWalkedModified - player.prevDistanceWalkedModified) * partialTicks
+        angle1 += sin((walkedDist) * 6.0f) * 32.0f * cameraYaw
+        if (player.isSneaking) {
+            angle1 += 25.0f
+        }
+
+        GlStateManager.rotate(6.0f + angle2 / 2.0f + angle1, 1.0f, 0.0f, 0.0f)
+        GlStateManager.rotate(angle3 / 2.0f, 0.0f, 0.0f, 1.0f)
+        GlStateManager.rotate(-angle3 / 2.0f, 0.0f, 1.0f, 0.0f)
+        GlStateManager.rotate(180.0f, 0.0f, 1.0f, 0.0f)
+
+        renderer.mainModel.renderCape(0.0625f)
+        GlStateManager.popMatrix()
     }
 
-    private fun formatUUID(uuid: String?): String {
-        return uuid!!.replace("-".toRegex(), "")
+    fun tryRenderElytra(
+            renderer: RenderLivingBase<*>,
+            model: ModelElytra,
+            entity: EntityLivingBase,
+            limbSwing: Float,
+            limbSwingAmount: Float,
+            ageInTicks: Float,
+            netHeadYaw: Float,
+            headPitch: Float,
+            scale: Float,
+            partialTicks: Float
+    ): Boolean {
+        if (isDisabled
+                || entity !is AbstractClientPlayer
+                || entity.getItemStackFromSlot(EntityEquipmentSlot.CHEST).getItem() != Items.ELYTRA) return false
+
+        val cape = capeUsers[entity.gameProfile.id]
+
+        return if (cape != null) {
+            renderElytra(renderer, model, entity, cape, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scale, partialTicks)
```

Comment:
I think we should inline it for less ♿ parameter spam

Reply:
mfw